packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -699,34 +685,18 @@ export class GarbageCollector implements IGarbageCollector {
 		// This is important because a container may never load during a node's Sweep Grace Period,
 		// so that node would directly become sweep-ready skipping over tombstone-ready state,
```

Comment:
> that node would directly become sweep-ready skipping over tombstone-ready state,

Just some thoughts: Ideally flags/states TombstoneReady/SweepReady should be stored in nodes; to avoid such risky scenarios. A node should not be able to skip a state like this!

Reply:
@markfields and I have discussd adding tombstone ready as a state in the node and ensure that it doesn't skip over states. However, it's not that simple to do it - We will have to store the sweep grace period in the summary and that changes the summary format. It's not a big deal but we have to handle compat and it will take time. So, we had punted doing this only if necessary.
We could add it to the GC backlog and do it opportunistically.

Reply:
Actually, we don't have to store the sweep grace period. For every node, we have to save the time it was marked tombstone which makes it quite expensive.

Reply:
It's already on the backlog AB#6652